### DISCUSSION TITLE
Debug option for video processing statistics

### DISCUSSION THREAD

**Discussion by Semmu** [Motion-Project/motion#1646] [General]
Hi,

It would be very handy to have motion report some statistics when it is processing a video feed. I'm thinking about making stats such as changed pixels and the number of consecutive motion-detected frames available and printed on each new frame.

Why? Basically for debug purposes: to see what motion "thinks" while processing a feed. It would be useful to see these stats for faster and more reliable tuning of e.g. the motion detection threshold values.

This way one could simply see what motion "thinks" of a video feed when nothing is happening (e.g. just wind blowing) and when there is some actual movement to be detected.

This setting would not be on by default, because it would probably mean a lot of overhead, but for initial tuning of config values, it would be very useful.

What do you think?

**Comment by Mr-Dave** [comment]
This sounds essentially the same as [setup_mode](https://motion-project.github.io/motion_config.html#setup_mode) and/or [stream_preview_method](https://motion-project.github.io/motion_config.html#stream_preview_method) option 3

**Reply by Semmu** [reply]
Didn't know about setup mode, but indeed it seems very similar, thanks! Will take a look.
